SignalServiceKit/src/Messages/Interactions/TSOutgoingMessage.m

```diff
@@ -478,15 +482,15 @@ - (void)addLocalProfileKeyIfNecessary:(OWSSignalServiceProtosContentBuilder *)co
         //
         // For Group threads, we want to include the profile key IFF the
         // recipient OR the group is in the whitelist.
-        if ([OWSProfilesManager.sharedManager isUserInProfileWhitelist:recipient.recipientId]) {
+        if ([profileManager isUserInProfileWhitelist:recipient.recipientId]) {
             shouldIncludeProfileKey = YES;
-        } else if ([OWSProfilesManager.sharedManager isThreadInProfileWhitelist:self.thread]) {
+        } else if ([profileManager isThreadInProfileWhitelist:self.thread]) {
             shouldIncludeProfileKey = YES;
         }
     }
 
     if (shouldIncludeProfileKey) {
-        [contentBuilder setProfileKey:OWSProfilesManager.sharedManager.localProfileKey];
+        [contentBuilder setProfileKey:localProfileKey];
```

Comment:
per earlier discussion this will be in the DataMessage, but we can do that in a followup.